cephadm-preflight.yml

```diff
@@ -41,284 +41,20 @@
       - ceph-radosgw
       - rbd-mirror
   tasks:
-    - name: import_role ceph_defaults
+    - name: Import_role ceph_defaults
       import_role:
         name: ceph_defaults
 
-    - name: redhat family of OS related tasks
-      when: ansible_facts['os_family'] == 'RedHat'
-      block:
-        - name: rhcs related tasks
-          when: ceph_origin == 'rhcs'
-          block:
-            - name: enable red hat ceph storage tools repository
-              rhsm_repository:
-                name: "rhceph-{{ ceph_rhcs_version }}-tools-for-rhel-{{ ansible_facts['distribution_major_version'] }}-{{ ansible_facts['architecture'] }}-rpms"
+    - name: Import specific OS-related tasks
```

Comment:
this doesn't seem to work

Reply:
```
TASK [Import specific OS-related tasks] ****************************************
task path: /home/jenkins-build/build/workspace/cephadm-ansible-prs-el9-functional/cephadm-preflight.yml:48
fatal: [ceph-node0]: FAILED! => 
  reason: Could not find or access '/home/jenkins-build/build/workspace/cephadm-ansible-prs-el9-functional/[]' on the Ansible Controller.
fatal: [ceph-node1]: FAILED! => 
  reason: Could not find or access '/home/jenkins-build/build/workspace/cephadm-ansible-prs-el9-functional/[]' on the Ansible Controller.
fatal: [ceph-node2]: FAILED! => 
  reason: Could not find or access '/home/jenkins-build/build/workspace/cephadm-ansible-prs-el9-functional/[]' on the Ansible Controller.
fatal: [ceph-node3]: FAILED! => 
  reason: Could not find or access '/home/jenkins-build/build/workspace/cephadm-ansible-prs-el9-functional/[]' on the Ansible Controller.
fatal: [ceph-node4]: FAILED! => 
  reason: Could not find or access '/home/jenkins-build/build/workspace/cephadm-ansible-prs-el9-functional/[]' on the Ansible Controller.
fatal: [ceph-node5]: FAILED! => 
  reason: Could not find or access '/home/jenkins-build/build/workspace/cephadm-ansible-prs-el9-functional/[]' on the Ansible Controller.
fatal: [client0]: FAILED! => 
  reason: Could not find or access '/home/jenkins-build/build/workspace/cephadm-ansible-prs-el9-functional/[]' on the Ansible Controller.
fatal: [client1]: FAILED! => 
  reason: Could not find or access '/home/jenkins-build/build/workspace/cephadm-ansible-prs-el9-functional/[]' on the Ansible Controller.

```

Reply:
why do we have to make it that 'complex' ?
I didn't test the following myself but I'd expect it to work as expected:
```yaml
      import_tasks: tasks/redhat.yml
      - name: Import specific OS-related tasks
        import_tasks: "{{ item }}"
        with_first_found:
          - "{{ ansible_distribution }}.yml"
```
any issue with it?

Reply:
I used the more complex way as when i did local linting it complained about import_tasks

```
WARNING  Listing 1 violation(s) that are fatal
syntax-check[specific]: You cannot use loops on 'import_tasks' statements. You should use 'include_tasks' instead.
cephadm-preflight.yml:48:7
```

Reply:
there was also this on the previous job run

```
Using /home/jenkins-build/build/workspace/cephadm-ansible-prs-el9-functional/tests/functional/ansible.cfg as config file
ERROR! the role 'ceph-defaults' was not found in /home/jenkins-build/build/workspace/cephadm-ansible-prs-el9-functional/roles:/home/jenkins-build/build/workspace/cephadm-ansible-prs-el9-functional:/home/jenkins-build/build/workspace/cephadm-ansible-prs-el9-functional

The error appears to be in '/home/jenkins-build/build/workspace/cephadm-ansible-prs-el9-functional/collect-logs.yml': line 6, column 15, but may
be elsewhere in the file depending on the exact syntax problem.

The offending line appears to be:

      import_role:
        name: ceph-defaults
              ^ here
```

Reply:
Here is some links I looked over :
-  [first_found_lookup](https://docs.ansible.com/ansible/latest/collections/ansible/builtin/first_found_lookup.html)
-  [Tasks - at the bottom of the page](https://richm.github.io/how-to-include-vars-and-tasks-in-ansible.html)

Reply:
> there was also this on the previous job run
> 
> ```
> Using /home/jenkins-build/build/workspace/cephadm-ansible-prs-el9-functional/tests/functional/ansible.cfg as config file
> ERROR! the role 'ceph-defaults' was not found in /home/jenkins-build/build/workspace/cephadm-ansible-prs-el9-functional/roles:/home/jenkins-build/build/workspace/cephadm-ansible-prs-el9-functional:/home/jenkins-build/build/workspace/cephadm-ansible-prs-el9-functional
> 
> The error appears to be in '/home/jenkins-build/build/workspace/cephadm-ansible-prs-el9-functional/collect-logs.yml': line 6, column 15, but may
> be elsewhere in the file depending on the exact syntax problem.
> 
> The offending line appears to be:
> 
>       import_role:
>         name: ceph-defaults
>               ^ here
> ```

you can ignore this error

Reply:
> I used the more complex way as when i did local linting it complained about import_tasks
> 
> ```
> WARNING  Listing 1 violation(s) that are fatal
> syntax-check[specific]: You cannot use loops on 'import_tasks' statements. You should use 'include_tasks' instead.
> cephadm-preflight.yml:48:7
> ```
> 
> Which it complained about in the latest job [here](https://2.jenkins.ceph.com/job/cephadm-ansible-prs-el9-functional/243/console)

yeh the linter error makes sense, then just use `include_tasks` instead

Reply:
I have ran this locally against my own inv file and don't get the same error than what is in the CI 

```
TASK [Import specific OS-related tasks] ****************************************
task path: /home/jenkins-build/build/workspace/cephadm-ansible-prs-el9-functional/cephadm-preflight.yml:48
fatal: [ceph-node0]: FAILED! => 
  reason: Could not find or access '/home/jenkins-build/build/workspace/cephadm-ansible-prs-el9-functional/[]' on the Ansible Controller.
fatal: [ceph-node1]: FAILED! => 
  reason: Could not find or access '/home/jenkins-build/build/workspace/cephadm-ansible-prs-el9-functional/[]' on the Ansible Controller.
fatal: [ceph-node2]: FAILED! => 
  reason: Could not find or access '/home/jenkins-build/build/workspace/cephadm-ansible-prs-el9-functional/[]' on the Ansible Controller.
fatal: [ceph-node3]: FAILED! => 
  reason: Could not find or access '/home/jenkins-build/build/workspace/cephadm-ansible-prs-el9-functional/[]' on the Ansible Controller.
fatal: [ceph-node4]: FAILED! => 
  reason: Could not find or access '/home/jenkins-build/build/workspace/cephadm-ansible-prs-el9-functional/[]' on the Ansible Controller.
fatal: [ceph-node5]: FAILED! => 
  reason: Could not find or access '/home/jenkins-build/build/workspace/cephadm-ansible-prs-el9-functional/[]' on the Ansible Controller.
fatal: [client0]: FAILED! => 
  reason: Could not find or access '/home/jenkins-build/build/workspace/cephadm-ansible-prs-el9-functional/[]' on the Ansible Controller.
fatal: [client1]: FAILED! => 
  reason: Could not find or access '/home/jenkins-build/build/workspace/cephadm-ansible-prs-el9-functional/[]' on the Ansible Controller.
```

Reply:
simplfying the include_tasks  shows a path, but still not available. 
```
    - name: Import specific OS-related tasks
      include_tasks: "tasks/{{ ansible_facts['distribution'] }}.yml"
```
I am wondering if the tasks folder path needs adjusting, as the `Vagrantfile` is using `    ansible.playbook = '../cephadm-preflight.yml'` 

```
TASK [Import specific OS-related tasks] ****************************************
task path: /home/jenkins-build/build/workspace/cephadm-ansible-prs-el9-functional/cephadm-preflight.yml:48
fatal: [ceph-node0]: FAILED! => 
  reason: Could not find or access '/home/jenkins-build/build/workspace/cephadm-ansible-prs-el9-functional/tasks/CentOS.yml' on the Ansible Controller.
fatal: [ceph-node1]: FAILED! => 
  reason: Could not find or access '/home/jenkins-build/build/workspace/cephadm-ansible-prs-el9-functional/tasks/CentOS.yml' on the Ansible Controller.
fatal: [ceph-node2]: FAILED! => 
  reason: Could not find or access '/home/jenkins-build/build/workspace/cephadm-ansible-prs-el9-functional/tasks/CentOS.yml' on the Ansible Controller.
fatal: [ceph-node3]: FAILED! => 
  reason: Could not find or access '/home/jenkins-build/build/workspace/cephadm-ansible-prs-el9-functional/tasks/CentOS.yml' on the Ansible Controller.
fatal: [ceph-node4]: FAILED! => 
  reason: Could not find or access '/home/jenkins-build/build/workspace/cephadm-ansible-prs-el9-functional/tasks/CentOS.yml' on the Ansible Controller.
fatal: [ceph-node5]: FAILED! => 
  reason: Could not find or access '/home/jenkins-build/build/workspace/cephadm-ansible-prs-el9-functional/tasks/CentOS.yml' on the Ansible Controller.
fatal: [client0]: FAILED! => 
  reason: Could not find or access '/home/jenkins-build/build/workspace/cephadm-ansible-prs-el9-functional/tasks/CentOS.yml' on the Ansible Controller.
fatal: [client1]: FAILED! => 
  reason: Could not find or access '/home/jenkins-build/build/workspace/cephadm-ansible-prs-el9-functional/tasks/CentOS.yml' on the Ansible Controller.
```